### PR TITLE
Exclude fields `status` and `creationTimestamp` from data center and machine configs

### PR DESCRIPTION
**Pull request by mitalipaygude**
Exclude fields `status` and `creationTimestamp` from data center and machine configs

*Issue #, if available:*
#191
Changes for Machine and Data center configs.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/vsphere/vsphere.go
Comment (jiayiwang7) [inline pkg/providers/vsphere/vsphere.go:1408]: whats the reason to use generated config here?

File: pkg/providers/vsphere/vsphere.go
Comment (jiayiwang7) [inline pkg/providers/vsphere/vsphere.go:1382]: @danbudris will the change here affect your PR on MachineConfigs? we remove the status field.

File: pkg/api/v1alpha1/vspheredatacenterconfig_types.go
Comment (jiayiwang7) [inline pkg/api/v1alpha1/vspheredatacenterconfig_types.go:88]: where is the function used?

File: pkg/api/v1alpha1/vspheredatacenterconfig_types.go
Comment (danbudris) [inline pkg/api/v1alpha1/vspheredatacenterconfig_types.go:65]: would this be better named `VsphereDatacenterConfigGenerated` with `Generated` instead of `Generate`? Maybe I'm misunderstanding

File: pkg/providers/vsphere/vsphere.go
Comment (danbudris) [reply pkg/providers/vsphere/vsphere.go:1382]: so in my PR [I'm adding the `OSFamily()` method to the `MachineConfig` interface](https://github.com/aws/eks-anywhere/pull/286/files#diff-8f631100c024f9035d740ab95c024634e695f200be9fde670b7a85a101d1b8edR51)

along with the [corresponding implementation in the `VSphereMachineConfig`](https://github.com/aws/eks-anywhere/pull/286/files#diff-88ba4e36f3f46665ba8a2a980719c15dca7167959b8bd04d787184402f39c602R52)

Comment (danbudris) [reply pkg/providers/vsphere/vsphere.go:1382]: so as long as the MachineConfigs being returned still implement the `MachineConfig` interface in the `Provider` package, we're all good -- I don't need access to status or kind even

Comment (jiayiwang7) [reply pkg/providers/vsphere/vsphere.go:1382]: roger. thanks!

File: pkg/api/v1alpha1/vspheredatacenterconfig_types.go
Comment (mitalipaygude) [reply pkg/api/v1alpha1/vspheredatacenterconfig_types.go:65]: Its named as `Generate` struct across all configs because the primary use of these structs was for the `generate clusterconfig` command. For consistency purposes, this struct is also named as `Generate` and not `Generated`.

**Comment by abhay-krishna** [comment]
@mitalipaygude: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-cluster-controller-tooling-presubmit | c3d20f11be78574d933903c495c03b07379ac08b | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/283/eks-anywhere-cluster-controller-tooling-presubmit/1441088063718559744) | `/test eks-anywhere-cluster-controller-tooling-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=283). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:mitalipaygude).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by jiayiwang7 — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/283#pullrequestreview-762398763" title="Approved">jiayiwang7</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/283#" title="Author self-approved">mitalipaygude</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jiayiwang7,mitalipaygude]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
@mitalipaygude: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-cluster-controller-tooling-presubmit | c3d20f11be78574d933903c495c03b07379ac08b | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/283/eks-anywhere-cluster-controller-tooling-presubmit/1441088063718559744) | `/test eks-anywhere-cluster-controller-tooling-presubmit`
eks-anywhere-attribution-files-presubmit | 9baeed56f5917780d71f764cc9694e08f974ff32 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/283/eks-anywhere-attribution-files-presubmit/1441139654395957248) | `/test eks-anywhere-attribution-files-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=283). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:mitalipaygude).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Pull request merged**
